### DISCUSSION TITLE
Customized messages

### DISCUSSION THREAD

**Discussion by tonnic** [berrystephenw/OctoText#12] [Ideas]
Is there a way to customize messages with variables?  I have multiple printers, so it would be nice to see which printer, as well as other variables that could be passed:

For example (in the subject/body of the text/email), allow variables such as:
- printer name
- filename
- time remaining
- current layer height
- total layer height
- current nozzle temp
- current bed temp
- current pi temp (soct)
- current fan speed

**Comment by berrystephenw** [comment]
Interesting ideas. Some of that is easier than others. I'm collecting feedback for the next revision and I'll put yours on the list. Printer name has been mentioned more than once... Thanks!

**Reply by garyriet** [reply]
Great thanks  I have 9 printers so printer name would be cool!  Awesome plugin! Thanks

**Comment by victaru** [comment]
honestly i'm surprised only one person had this request...customised email subject eg: *gcode* file printing status would be awesome! this plugin really helps alot :)

**Reply by berrystephenw** [reply]
I understand the request, and if it were easy, I'd certainly consider putting it in. I think this would add a lot more forms, string parsing, error checking etc.... 
at the moment I'm focused on making OT more robust (retrying temporary network failures) and adding a simple API interface. The API would allow another plug-in to form messages however they want for whatever event they choose. 
I'll release that on the rc channel soon along with a skeleton plug-in as an example.
